src/stories/Forms/Toggle/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { GlobalInputProps } from 'src';
+import styled from 'styled-components';
+
+const Wrapper = styled.div`
+  height: var(--spacing-24);
+  width: 44px;
+  position: relative;
+`;
+
+const StyledInput = styled.input`
+  // the toggle button uses a checkbox for UU reasons
+  // this is then restyled to look like a toggle button
+  width: 100%;
+  height: 100%;
+  border-radius: 20px;
+  background: var(--color-grey55);
+  appearance: none;
+  outline: none;
+
+  // the before element is adding the circle to the toggle button
+  ::before {
+    height: 20px;
+    width: 20px;
+    position: absolute;
+    top: 2px;
+    left: 2px;
+    border-radius: 50%;
+    background: white;
+    transition: 0.2s ease-out;
+    content: '';
+  }
+
+  :checked {
+    background: var(--color-lavender);
+  }
+
+  :checked::before {
+    left: 22px;
+  }
+
+  :checked:hover {
+    background: var(--color-salviaBlue);
+  }
+
+  :hover {
+    background: var(--color-grey45);
+  }
+
+  :checked:disabled {
+    background: var(--color-plumbagoBlue);
+  }
+
+  :disabled {
+    background: var(--color-grey76);
+  }
+
+  :focus-visible {
+    outline-style: auto;
+    outline-color: var(--color-salviaBlue);
+    outline-offset: 3px;
+  }
```

Comment:
Missing hasError state. 
It may never occur that a toggle should be required or fail etc.., but it doesn't hurt to support the prop as it's a part of globalInputProps.

On another note, great job keeping track of all these states and colors. I could never 😅